jbmc/src/java_bytecode/java_trace_validation.cpp

```diff
@@ -0,0 +1,189 @@
+/*******************************************************************\
+
+Module: Java trace validation
+
+Author: Jeannie Moulton
+
+\*******************************************************************/
+
+#include "java_trace_validation.h"
+
+#include <algorithm>
+
+#include <goto-programs/goto_trace.h>
+#include <util/expr.h>
+#include <util/expr_util.h>
+#include <util/simplify_expr.h>
+#include <util/std_expr.h>
+
+bool check_symbol_structure(const exprt &symbol_expr)
+{
+  const auto symbol = expr_try_dynamic_cast<symbol_exprt>(symbol_expr);
+  return symbol && !symbol->get_identifier().empty();
+}
+}
```

Comment:
:no_entry_sign: This commit "Add trace validation for for jbmc" doesn't look like it will compile (have I mentioned how much I hate rebasing)  - could you fix it